### PR TITLE
Add music and sound!!!

### PR DESCRIPTION
**Pull request by caiges**
- Add an audio player for playing a background music track and sound effects.
- Shuffle where our subsystems are initialized so they're accessible from the game type methods.
- Added the notion of "asset packs" for audio and graphical assets so that we can easily test and select different assets.

### CODE REVIEW

File: src/audio/player.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+impl<'a> Player<'a> {
+  pub fn new(pack: String) -> Player<'a> {
+    let mut catalog: HashMap<String, mixer::Chunk> = HashMap::new();
+    let mut music_catalog: HashMap<String, mixer::Music> = HashMap::new();
+    // We should make loading dynamic.
+    let pack_path = format!("assets/{}", pack);
+
+    // Load sound effects.
+    let audio_pack = [
```
Comment (jimbobhickville) [inline src/audio/player.rs]: Why not just use a HashMap?

**Review by jimbobhickville — APPROVED** [review]
lgtm. Just that one minor suggestion.

File: src/audio/player.rs (hunk 1 above)
Comment (caiges) [reply src/audio/player.rs]: Why not indeed, will fix.

Comment (caiges) [reply src/audio/player.rs]: Patched that up. Nice catch.

**Pull request merged**